src/daemons/linuxd/src/linuxd/mod.rs

```diff
@@ -189,10 +183,23 @@ impl LinuxDaemon {
                         Err(error) => {
                             warn!("failed to join new virtual environment (error={error:?})");
                             let message: Message = crate::build_error(source, error.code);
-                            uvm_handle
-                                .get_user_vm_stream()
-                                .lock()
-                                .unwrap()
+
+                            let uvm_stream: Arc<Mutex<(SocketStream, VecDeque<u8>)>> =
+                                uvm_handle.get_user_vm_stream();
+                            let mut guard: MutexGuard<'_, (SocketStream, VecDeque<u8>)> =
+                                match uvm_stream.lock() {
+                                    Ok(guard) => guard,
+                                    Err(e) => {
+                                        error!(
+                                            "error acquiring lock on user VM stream (error={e:?})"
+                                        );
+                                        continue;
```

Comment:
Failing to notify the user VM about this may have undefined behavior. If we fail to obtain the lock, it is likely that we had a contention, we could sleep and retry. If we retried for more than X times, than we should simply kill that VM (instead of letting it hanging). I understand that this is out of the scope of this PR. Could you create an issue and log it here?

Reply:
If I understand correctly, `lock()` will only return an error if the mutex is poisoned, which means that another thread panicked while holding the lock.

If this happens this means that something is going very wrong, so maybe we should just error-out? (It is unlikely that we will be able to clean this user VM anyway, neither).